embassy-stm32/src/dma/mod.rs

```diff
@@ -143,6 +143,13 @@ pub trait Word: sealed::Word {
     fn bits() -> WordSize;
 }
 
+impl sealed::Word for i8 {}
```

Comment:
I see, if the bit size is coupled to the type then it makes sense to stick to that, not add it in `Config`. Using `i8` is not nice, it's not intuitive and we're out of options if we want to add 6bit, 7bit, etc in the future. It might be better to define a new type for a 5-bit word:

`#[repr(transparent)] pub struct U5(pub u8);`

Reply:
Created an U8 type and this type selects a word_size of 5 bits. Placed NoDma in the test case as I did not saw a possibility to remove RxDma from the constructor.
A wish: Would be nice if the public constructor does support Some() for all pins and channels. Then less constructors are needed, as the user can make its own selection. Freedom to the masses!

Reply:
`Option<impl XxxPin<..>>` doesn't work well with type inference with the pin trait setup. Also, different constructors allow tracking the SPI capabilities in the type so that it only impls `SpiBusWrite` or `SpiBusRead`. (this is not implemented in nRF though)

Reply:
> I did not saw a possibility to remove RxDma from the constructor.

Should be doable if you put it in a different `impl` block that fixes RxDma=NoDma instead of being generic.